src/context.c

```diff
@@ -59,44 +44,51 @@ static gchar* get_boot_id(void)
 	return g_strchomp(contents);
 }
 
-static gchar* get_cmdline_bootname(void)
+static gchar *get_cmdline(void)
 {
-	g_autofree gchar *contents = NULL;
-	g_autofree gchar *realdev = NULL;
-	gchar *bootname = NULL;
+	gchar *contents = NULL;
+	g_autoptr(GError) ierror = NULL;
 
 	if (context->mock.proc_cmdline)
-		contents = g_strdup(context->mock.proc_cmdline);
-	else if (!g_file_get_contents("/proc/cmdline", &contents, NULL, NULL))
+		return g_strdup(context->mock.proc_cmdline);
+
+	if (!g_file_get_contents("/proc/cmdline", &contents, NULL, &ierror)) {
+		g_warning("Failed to get kernel cmdline: %s", ierror->message);
 		return NULL;
+	}
+
+	return contents;
+}
 
-	if (strstr(contents, "rauc.external") != NULL) {
+static gchar* get_cmdline_bootname_explicit(const gchar *cmdline)
+{
+	gchar *bootname = NULL;
+
+	if (strstr(cmdline, "rauc.external") != NULL) {
 		g_message("Detected explicit external boot, ignoring missing active slot");
 		return g_strdup("_external_");
 	}
 
-	bootname = regex_match("rauc\\.slot=(\\S+)", contents);
+	if (strstr(cmdline, "root=/dev/nfs") != NULL) {
+		g_message("Detected nfs boot, ignoring missing active slot");
+		return g_strdup("_external_");
+	}
+
+	bootname = r_regex_match_simple("rauc\\.slot=(\\S+)", cmdline);
 	if (bootname)
 		return bootname;
 
-	/* For barebox, we check if the bootstate code set the active slot name
-	 * in the command line */
-	if (g_strcmp0(context->config->system_bootloader, "barebox") == 0) {
-		bootname = regex_match(
-				"(?:bootstate|bootchooser)\\.active=(\\S+)",
-				contents);
-		if (bootname)
-			return bootname;
-	}
+	return bootname;
```

Comment:
```suggestion
	return NULL;
```

might be more explicit here.